web/status.go

```diff
@@ -0,0 +1,238 @@
+	// Memory stats for isvcs don't work the same way, so we don't append
+	// those services to this list yet
+	memoryStats := make(map[string]metrics.MemoryUsageStats)
+	if len(instances) > 0 {
+		response := []metrics.MemoryUsageStats{}
+		query := dao.MetricRequest{
+			StartTime: time.Now().Add(-time.Hour * 24),
+			Instances: convertInstancesToMetric(instances),
+		}
+		if err := client.GetInstanceMemoryStats(query, &response); err != nil {
+			glog.Errorf("Unable to look up instance memory stats (%s)", err)
+		}
+		for _, mus := range response {
+			memoryStats[memoryKey(mus.ServiceID, mus.InstanceID)] = mus
+		}
+	}
+
+	// Look up health check statuses
+	healthChecks := make(map[string]map[string]health.HealthStatus)
+	if len(instances) > 0 {
+		results := make(map[string]map[int]map[string]health.HealthStatus)
+		if err := client.GetServicesHealth(0, &results); err != nil {
+			glog.Errorf("Unable to look up health check results (%s)", err)
+		}
+		for svcid, insts := range results {
+			for instid, checks := range insts {
+				healthChecks[memoryKey(svcid, string(instid))] = checks
+			}
+		}
+	}
+
+	// Add isvcs to the mix
+	isvcInstances := getIRS()
+	for _, isvc := range isvcInstances {
+		var checks map[string]health.HealthStatus
+		instances = append(instances, isvc)
+		if isvc.ServiceID == "isvc-internalservices" {
+			checks = isvcsRootHealth
+		} else {
+			results, err := isvcs.Mgr.GetHealthStatus(strings.TrimPrefix(isvc.ServiceID, "isvc-"))
+			if err != nil {
+				glog.Warningf("Error acquiring health status for %s (%s)", isvc.ServiceID, err)
+				continue
+			}
+			checks = convertDomainHealthToNewHealth(results.HealthStatuses)
+		}
+		healthChecks[memoryKey(isvc.ServiceID, string(isvc.InstanceID))] = checks
+	}
+	// Create the concise service statuses
+	for _, instance := range instances {
+		stat := &ConciseServiceStatus{
+			DockerID:        instance.DockerID,
+			HostID:          instance.HostID,
+			ID:              instance.ID,
+			InSync:          instance.InSync,
+			InstanceID:      instance.InstanceID,
+			Instances:       instance.Instances,
+			Name:            instance.Name,
+			ParentServiceID: instance.ParentServiceID,
+			PoolID:          instance.PoolID,
+			ServiceID:       instance.ServiceID,
+			StartedAt:       instance.StartedAt,
+			RAMCommitment:   instance.RAMCommitment,
+		}
+		key := memoryKey(instance.ServiceID, string(instance.InstanceID))
+		if mem, ok := memoryStats[key]; ok {
+			stat.RAMMax = mem.Max
+			stat.RAMLast = mem.Last
+			stat.RAMAverage = mem.Average
+		}
+		if hc, ok := healthChecks[key]; ok {
+			stat.HealthChecks = hc
+		}
+		statuses = append(statuses, stat)
+	}
+	return
+}
+
+func restGetConciseServiceStatus(w *rest.ResponseWriter, r *rest.Request, client *node.ControlClient) {
+	f := func() []byte {
+		statuses, err := getAllServiceStatuses(client)
+		if err != nil {
+			glog.Errorf("Error retrieving service statuses: (%s)", err)
+			restServerError(w, err)
```

Comment:
I think we're going to hit a runtime error in this case - at a minimum the outcome is somewhat ambiguous. 

Consider the scenario where the cached value is empty, and the system calls `f` which then calls `restServerError()`. `restServerError()` is going to write an error payload to `w`, then `f` is going to continue, and marshal a nil value for `statuses` (which might panic?) and then return a nil value which then is passed to `w.Write()` which will do what? maybe overwrite the error payload that  `restServerError()` wrote? IDK.

Seems like it would be cleaner to have `f()` return an `error` as well as `[]byte` just to eliminate the ambiguity.